Library/Service/SPITenantsService.m

```diff
@@ -0,0 +1,59 @@
+//
+//  SPITenantsService.m
+//  SPIClient-iOS
+//
+//  Created by mx51 on 25/02/21.
+//  Copyright © 2021 mx51. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "SPITenantsService.h"
+
+static NSTimeInterval connectionTimeout = 8; // HTTP connection timeout
+
+@implementation SPITenantsService : NSObject
+
+- (void)retrieveTenants:(NSString *)posVendorId
+                 apiKey:(NSString *)apiKey
+            countryCode:(NSString *)countryCode
+             completion:(SPITenantsResult)completion {
+    NSString *tenantServiceUrl = [NSString stringWithFormat: @"https://spi.integration.mspenv.io/tenants?country-code=%@&pos-vendor-id=%@&api-key=%@", countryCode, posVendorId, apiKey];
+    
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
```

Comment:
It's bad practice to manually alloc and init built-in classes, use one of the helpers such as `requestWithURL`.